apps/astarte_data_updater_plant/test/astarte_data_updater_plant/cache_test.exs

```diff
@@ -0,0 +1,202 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.DataUpdaterPlant.DataUpdater.CacheTest do
+  use ExUnit.Case, async: true
+  use ExUnitProperties
+  import Mimic
+
+  alias Astarte.DataUpdaterPlant.DataUpdater.Cache
+
+  setup :verify_on_exit!
+
+  describe "new/1" do
+    test "returns a new cache tuple with specified size" do
+      cache = Cache.new(3)
+      assert cache == {3, %{}}
```

Comment:
nit: I think that testing this might be testing an implementation detail. If tomorrow the cache implementation changes this test has to be rewritten. I think the scope of `new/1` is no narrow that we don't actually need this test